Userland/Libraries/LibWeb/HTML/Scripting/ImportMapParseResult.cpp

```diff
@@ -41,23 +40,35 @@ JS::NonnullGCPtr<ImportMapParseResult> ImportMapParseResult::create(JS::Realm& r
     return result;
 }
 
-void ImportMapParseResult::visit_host_defined_self(JS::Cell::Visitor& visitor)
+void ImportMapParseResult::visit_host_defined_self(Visitor& visitor)
 {
     visitor.visit(*this);
 }
 
 void ImportMapParseResult::visit_edges(Visitor& visitor)
 {
     Base::visit_edges(visitor);
-    visitor.visit(m_error_to_rethrow);
+    if (m_error_to_rethrow.has_value()) {
+        m_error_to_rethrow.value().visit(
+            [&](WebIDL::SimpleException const&) {
+                // ignore
+            },
+            [&](JS::NonnullGCPtr<WebIDL::DOMException> exception) {
+                visitor.visit(exception);
+            },
+            [&](JS::Completion const&) {
+                // ignore
+            });
```

Comment:
Is this correct? What if something throws e.g a `JS::TypeError`? Then we'd need to visit it, I would think.

Reply:
Hmm, I'll take another look at this tonight 🤔

Reply:
I've amended this to
```cpp
    if (m_error_to_rethrow.has_value()) {
        m_error_to_rethrow.value().visit(
            [&](WebIDL::SimpleException const&) {
                // ignore
            },
            [&](JS::NonnullGCPtr<WebIDL::DOMException> exception) {
                visitor.visit(exception);
            },
            [&](JS::Completion const& completion) {
                if (completion.value().has_value())
                    visitor.visit(completion.value().value());
            });
    }
```

:^)